modules/mbedtls/Kconfig.tls-generic

```diff
@@ -105,7 +105,7 @@ config MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 
 config MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED
 	bool "ECDH-ECDSA based ciphersuite modes"
-	depends on MBEDTLS_ECDH_C && MBEDTLS_ECDSA_C
+	depends on (MBEDTLS_ECDH_C && MBEDTLS_ECDSA_C) || (PSA_WANT_ALG_ECDH && PSA_WANT_ALG_ECDSA)
```

Comment:
Are we gonna have other MbedTLS Kconfig options start depending on PSA ones? :thinking:

Reply:
Would it be more convenient (at least for the reader) to try to emulate the "CAN"-concept from Mbed TLS as Kconfigs?

Like `MBEDTLS_CAN_ECDH`
https://github.com/Mbed-TLS/mbedtls/blob/2ca6c285a0dd3f33982dd57299012dacab1ff206/include/mbedtls/config_adjust_legacy_crypto.h#L309C9-L309C25

Or more distinct: `MBEDTLS_PK_CAN_ECDSA_SOME`
https://github.com/Mbed-TLS/mbedtls/blob/2ca6c285a0dd3f33982dd57299012dacab1ff206/include/mbedtls/config_adjust_legacy_crypto.h#L333C9-L333C34

There are more than just `PSA_WANTS_ALG_XXXX` that can be used as a signal, since you can choose to use or not use PSA crypto with `MBEDTLS_USE_PSA_CRYPTO`. Arguably we of course prefer the `MBEDTLS_USE_PSA_CRYPTO` and PSA crypto will at some point likely be the only option, but it is still slightly more readable no not have to rehash both legacy and PSA configs for every ciphersuite

When I start thinking about it, there is an option to have a stable configuration follow the pattern `MBEDTLS_CAN_ECDH_ECDSA_KEY_EXCHANGE` for every relevant key exchange/cipher-suite

We have made a similar concept in nRF Connect SDK

Reply:
TBH in my first attempt I was about to remove those "depends on" because this is totally duplicating checks that MbedTLS already performs in `check_config.h` and, in case something is missing, it already prints a clear message of what's missing and where to look at.
Honestly I'm not a huge fan of duplicating also the `MBEDTLS_CAN_ECDH` and other MbedTLS internal symbols, because this adds extra burden of work every time Zephyr updates the MbedTLS library because you need to update all these checks again and again.
In my opinion I would follow the same path I've introduced for the `PSA_WANT` symbols: just add 1 Kconfig symbol for every MbedTLS build symbol, but no extra logic. Then if something is not correct `check_config.h` will error out some message. No extra work for Zephyr. However this proposed approach imposes quite some change to the current `Kconfig.tls-generic" file so I would keep this for a follow up PR in case we find an agreement (@tomi-font ).
Wdyt?

Reply:
Agreed to keep further significant work for a follow-up PR. And at first thought it seems senseful to get rid of `depends on` statements if those checks are performed by MbedTLS itself.

Reply:
This is for sure follow-up work. This only makes sense if we have stable definitions in Mbed TLS distribution. Arguably PSA_WANT_ALG_XXXX should be sufficient for most cases, but then this can only happen when mbedtls_ prefixed crypto toolbox APIs are removed from the system (e.g. in downstream Zephyr or in upstream Mbed TLS)